src/elements/content-sidebar/activity-feed/task-new/AvatarGroup.js

```diff
@@ -0,0 +1,127 @@
+// @flow strict
+import * as React from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import type { InjectIntlProvidedProps } from 'react-intl';
+import uniqueId from 'lodash/uniqueId';
+import { Flyout, Overlay } from '../../../../components/flyout';
+import Tooltip from '../../../../components/tooltip';
+import PlainButton from '../../../../components/plain-button';
+import ReadableTime from '../../../../components/time/ReadableTime';
+import messages from '../../../common/messages';
+import AssigneeAvatar from './AssigneeAvatar';
+import { TASK_NEW_APPROVED, TASK_NEW_REJECTED, TASK_NEW_COMPLETED, TASK_NEW_NOT_STARTED } from '../../../../constants';
+
+import './AvatarGroup.scss';
+
+const MAX_AVATARS = 3;
+
+type Props = {|
+    assignees: TaskAssigneeCollection,
+    getAvatarUrl: GetAvatarUrlCallback,
+    maxAvatars: number,
+|} & InjectIntlProvidedProps;
+
+const statusMessages = {
+    [TASK_NEW_APPROVED]: messages.tasksFeedStatusApproved,
+    [TASK_NEW_REJECTED]: messages.tasksFeedStatusRejected,
+    [TASK_NEW_COMPLETED]: messages.tasksFeedStatusCompleted,
+    [TASK_NEW_NOT_STARTED]: null,
+};
+
+const Datestamp = ({ date }: { date: ISODate | Date }) => {
+    return <ReadableTime timestamp={new Date(date).getTime()} alwaysShowTime relativeThreshold={0} />;
+};
+
+const AvatarDetails = React.memo(({ user, status, completedAt, className }) => {
+    const statusMessage = statusMessages[status] || null;
+    return (
+        <div className={className}>
+            <div className="bcs-avatar-group-details-name">{user.name}</div>
+            {statusMessage && completedAt && (
+                <div className="bcs-avatar-group-details-status">
+                    <FormattedMessage {...statusMessage} values={{ dateTime: <Datestamp date={completedAt} /> }} />
+                </div>
+            )}
+        </div>
+    );
+});
+
+class AvatarGroup extends React.Component<Props> {
+    static defaultProps = {
+        maxAvatars: MAX_AVATARS,
+    };
+
+    listTitleId = `avatar-group-list-title-${uniqueId()}`;
+
+    render() {
+        const { maxAvatars, assignees = {}, getAvatarUrl, intl } = this.props;
```

Comment:
Seems like `assignees` should have a default prop value, instead.

Reply:
Done, and renamed to be more generic (`users`) in the event we extract this out as a separate reusable component.